### ISSUE TITLE
Enable user confirmation when Quit/Close window is triggered (CTRL+SHIFT+W)

### ISSUE THREAD

**Issue by krstp** [zed-industries/zed#4322]
### Check for existing issues

- [X] Completed

### Describe the feature

At times when working on multiple projects/windows, instead of hitting `CMD+W` to close a single pane, I press `CMD+SHIFT+W` which closes whole project window. At times this becomes troublesome. What I would like to request is additional prompt (pop-up) requiring user to confirm window (hence all panes) closure.

### If applicable, add mockups / screenshots to help present your vision of the feature

Here is a sample of what I use in Sublime Text, at least "Quit Now" would suffice; something along those lines or pop-up with OK/Cancel choice.

![Screenshot 2023-11-27 at 6 37 25 PM](https://github.com/zed-industries/community/assets/4711498/7bb5404f-51c8-4e96-92ba-c95b33b596fa)

**Comment by danielweck** [comment]
Good point, I agree. I enabled "confirm quit" and it works fine with CMD/CTRL+Q, but indeed for all intents and purposes, CMD/CTRL+SHIT+W has the same effect on an individual workspaces (in other words, could you please harmonise "confirm: quit app entirely" and "confirm: close this workspace while keeping others open"?).

PS: I am gradually transitioning from Visual Studio Code for some of my development projects (I love Zed's launch speed and general superior performance, so much so that I can live with a few inconsistencies / learning curve due to different design choices, missing features and bugs 👍 ). I was/am used to CMD/CTRL+W closing the VSCode workspace window after all individual file tabs are closed. It is strange to me that an additional shortcut with the SHIFT key modifier is necessary to achieve this in Zed, though I totally appreciate how useful CMD/CTRL+SHIT+W is when one or more file tab is still open (saves me hitting CMD/CTRL+W multiple times until the workspace window finally closes).

**Comment by danielweck** [comment]
Related issue: 
- https://github.com/zed-industries/zed/issues/7200

**Comment by notpeter** [comment]
This is now configurable with the following settings.

```json
  "confirm_quit": true,
```

Thanks

**Comment by enesbala5** [comment]
> This is now configurable with the following settings.
> 
>   "confirm_quit": true,
> 
> Thanks

This is not working for me unless I manually run `zed: quit` on the command palette. Quitting the app through SUPER + Q (on Linux) just closes it instantly, unlike VS Code which displays a confirmation popup instead.

**Comment by avinmaster** [comment]
Same ^

**Comment by karmeye** [comment]
There is still no config to warn before closing a _window_?

**Comment by enesbala5** [comment]
This issue needs to be re-opened

**Comment by tabulius-ux** [comment]
> There is still no config to warn before closing a _window_?

It does not work for me.

**Comment by karmeye** [comment]
> > There is still no config to warn before closing a _window_?
> 
> It does not work for me.

There is still no setting. `confirm_quit` is for application quit, not window close
